qiskit_nature/second_q/algorithms/excited_states_solvers/qeom.py

```diff
@@ -46,28 +54,55 @@ class QEOM(ExcitedStatesSolver):
     def __init__(
         self,
         ground_state_solver: GroundStateSolver,
-        excitations: Union[str, List[List[int]]] = "sd",
+        excitations: str
+        | int
+        | list[int]
+        | Callable[
+            [int, tuple[int, int]],
+            list[tuple[tuple[int, ...], tuple[int, ...]]],
+        ] = "sd",
     ) -> None:
         """
         Args:
             ground_state_solver: a GroundStateSolver object. The qEOM algorithm
                 will use this ground state to compute the EOM matrix elements
             excitations: The excitations to be included in the eom pseudo-eigenvalue problem.
-                If a string ('s', 'd' or 'sd') then all excitations of the given type will be used.
-                Otherwise a list of custom excitations can directly be provided.
+
+                :`str`: which contains the types of excitations. Allowed characters are
+                    + `s` for singles
+                    + `d` for doubles
+                :`int`: a single, positive integer which denotes the number of excitations
+                    (1 == `s`, etc.)
+                :`list[int]`: a list of positive integers generalizing the above
```

Comment:
Can this be detailed with an example like for the int case? 
My understanding is that the int type can be used to specify only excitations of degree n whereas the list can be used to specify multiple degrees.

```
Proposal:
:`int`: a single, positive integer which denotes one number of excitations (1 == `s`, 2== `d`, etc.)
:`list[int]`: a list of positive integer generalizing the above to multiple numbers of excitations ([1, 2]== `sd`, etc.)
```

Reply:
Sure, we can do that :+1: I simply copied the docstring from the excitation generator methods and `UCC`-style ansatze. I will update this according to your suggestion in all locations :+1: